tests/test_union.py

```diff
@@ -327,6 +327,44 @@ class A:
     assert from_dict(A, a_int_dict) == a_int
 
 
+def test_generic_union():
+    T = TypeVar('T')
+    U = TypeVar('U')
+
+    @serde
+    class Foo(Generic[T]):
+        v: T
+
+    @serde
+    class Bar(Generic[U]):
+        v: U
+
+    @serde
+    class A:
+        v: Union[Foo[int], Bar[str]]
+
+    a = A(Foo[int](10))
+    assert a == from_dict(A, to_dict(a))
+    assert a == from_tuple(A, to_tuple(a))
+
+    a = A(Foo[str]("foo"))
+    assert a == from_dict(A, to_dict(a))
+    assert a == from_tuple(A, to_tuple(a))
+
+    @serde
+    class B(Generic[T, U]):
+        v: Union[Foo[T], Bar[U]]
+
+    b = B[int, str](Foo[int](10))
+    assert b == from_dict(B[int, str], to_dict(b))
+    assert b == from_tuple(B[int, str], to_tuple(b))
+
+    b = B[Foo[int], Bar[str]](Foo(Foo(10)))
+    assert {'v': {'v': {'v': 10}}} == to_dict(b)
+    # TODO Nested parameterized generics still doesn't work
```

Comment:
Stale TODO? This test is passing

Reply:
This assert is supposed to be `==` not `!=`.

I will fix it to `==` and comment out.
```python
# assert b == from_dict(B[Foo[int], Bar[str]], to_dict(b))
```